Add `baffling-birthdays` exercise (#2539)
exercism/problem-specifications: exercises/baffling-birthdays/introduction.md

# Introduction

> Fresh out of graduation, you're throwing a huge party to celebrate with friends and family.

Reviewer: out of college/after graduation

Reviewer 2: Would `fresh out of school` be confusing to folks? That’s how I would say it vernacularly.

Over 70 people have shown up, including your mildly eccentric Uncle Ted.

> In one of his usual antics, he bets you £100 that at least two people in the room share their birthdays.

Reviewer 3: Consistency
```suggestion
In one of his usual antics, he bets you £100 that at least two people in the room share the same birthday.
```

> That sounds ridiculous—there are 365 possible birthdays, so you confidently accept.

Reviewer 2: ```suggestion
That sounds ridiculous—there are a lot of possible birthdays, so you confidently accept.
```
Leap years have 366 possible birthdays so I don't think we should specify a number. Referring to the count as some sort of large number still conveys the point that it's surprising the person found a match so quickly.

Author: I've rephrased

> That sounds ridiculous—there are 365 possible birthdays, so you confidently accept.

Reviewer: ```suggestion
That sounds ridiculous — there are 365 possible birthdays, so you confidently accept.
```

Reviewer 2: Is that a hyphen or an em dash? I’m on my phone currently so I can’t tell, but generally I’d expect an em dash (the long dash) to be used to conjoin independent clauses. I think spaces or no spaces around the em dash depends on the style guide you use. I haven’t had to write a term paper in over a decade so I’m not sure which style guide would be appropriate if we had to pick one.

Reviewer 3: It looks like an em dash. Spaces around it make it clearer that it's not a hyphen.

To your astonishment, after collecting just 32 birthdays, you've already found a match.

> Magnanimous, you hand Uncle Ted his £100, but something feels off.

Reviewer: Magnanimous or Magnanimously?

Reviewer 2: I think the wording is right since magnanimous as an adjective is describing the subject of the sentence (you) rather than the verb (hand). We might want to drop it though to avoid taking the reader out of the story.

Reviewer 3: If the sentence is kept, i'd rather drop the word and use something that's easier to understand
```suggestion
Accepting your loss, you hand Uncle Ted his £100, but something feels off.
```

The next day, curiosity gets the better of you.

> A quick web search leads you to the [Birthday Problem][birthday-problem], which reveals that with just 23 people, the probability of a shared birthday exceeds 50%.

Reviewer 2: ```suggestion
A quick web search leads you to the [birthday paradox][birthday-problem], which reveals that with just 23 people, the probability of a shared birthday exceeds 50%.
```

We should clearly refer to this as the birthday paradox. That'll align with how the Wikipedia article distinguishes the birthday problem and the paradox. Later on we also refer to "other paradoxes" without introducing a paradox so that's awkwardly written without this change.

Ah. So _that's_ why Uncle Ted was so confident.

> Determined to turn the tables, you start looking up other paradoxes—next time, _you'll_ be the one making the bets.

Reviewer: ```suggestion
Determined to turn the tables, you start looking up other paradoxes; next time, _you'll_ be the one making the bets.
```

~~~~exercism/note

> The birthday paradox is a veridical paradox: even though it feels wrong, it is actually true.

Reviewer 2: ```suggestion
The birthday paradox is a [veridical paradox][veridical-paradox]: even though it feels wrong, it is actually true.
```

> 
> [veridical-paradox]: https://en.wikipedia.org/wiki/Paradox#Quine's_classification

Reviewer 3: i believe this should no be inside the note?
```suggestion
```

Author: Which it is?

Reviewer 3: Apologies for the two separate comments.

Reviewer 3: > Which it is?

Both suggestions are valid. Remove from the note, and move at the end of the file

~~~~

> [birthday-problem]: https://en.wikipedia.org/wiki/Birthday_problem

Reviewer 3: ```suggestion
[birthday-problem]: https://en.wikipedia.org/wiki/Birthday_problem
[veridical-paradox]: https://en.wikipedia.org/wiki/Paradox#Quine's_classification
```

> [veridical-paradox]: https://en.wikipedia.org/wiki/Paradox#Quine's_classification
> ~~~~
> 
> [birthday-problem]: https://en.wikipedia.org/wiki/Birthday_problem

Reviewer 3: This was the correct suggestion. Ignore if the link renders correctly when inside the codeblock.
```suggestion
~~~~

[birthday-problem]: https://en.wikipedia.org/wiki/Birthday_problem
[veridical-paradox]: https://en.wikipedia.org/wiki/Paradox#Quine's_classification
```

Author: It does not only render correctly when it is in the note, but it has to be inside the note (for reasons)

Reviewer 3: Thanks for letting me know, I wasn't aware of this.
